src/main/java/TicTacToe/Board.java

```diff
@@ -88,10 +90,19 @@ public boolean hasTie() {
     }
 
     private boolean isFull() {
-        return getLines().stream().allMatch(line -> line.isFull());
+        return getLines().stream().allMatch(Line::isFull);
     }
 
     private int getSide() {
         return (int)Math.sqrt(grid.size());
     }
+
+    public void mark(String symbol, int position) {
```

Comment:
Is mark a symbol? I feel like you are using the same word to describe the something thing.
Also, you are using a low level data structure for the Mark, maybe it could be abstract with a Class or a Enum